### PR TITLE
Add OSSKU awareness to CSE and publishing-info

### PR DESCRIPTION
**Pull request by hbeberman**
Add OSSKU to NodeBootstrappingConfiguration
Update IsMariner and IsUbuntu checks to use NodeBootstrappingConfiguration
Pass OS_SKU into VHD publishing info as offer_name
Update OS_SKU for Mariner to CBLMariner and OS_VERSION to V1

### CODE REVIEW

File: vhdbuilder/packer/generate-vhd-publishing-info.sh [hunk 1]
```diff
@@ -39,7 +41,9 @@ cat <<EOF > vhd-publishing-info.json
     "sku_name" : "$sku_name",
     "offer_name" : "$OFFER_NAME",
     "hyperv_generation": "${HYPERV_GENERATION}",
-    "image_version": "${IMAGE_VERSION}"
+    "image_version": "${IMAGE_VERSION}",
+    "os_sku" : "${OS_SKU}",
+    "os_version" : "${OS_VERSION}"
```
Comment (xuto2) [inline vhdbuilder/packer/generate-vhd-publishing-info.sh:46]: do you have an example of generated content for ubuntu? Want to make sure there is not much confusion between os_sku and sku_name, os_version etc.

**Review by xuto2 — APPROVED** [review]

File: vhdbuilder/packer/generate-vhd-publishing-info.sh (hunk 1 above)
Comment (hbeberman) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:46]: Here's an example of an Ubuntu JSON. The os_sku and and os_version mostly stay the same (except 1604), and the sku_name is still version number concatenated with feature flags.
```json
{
    "vhd_url" : "<URL>",
    "os_name" : "Linux",
    "sku_name" : "1804containerd",
    "offer_name" : "Ubuntu",
    "hyperv_generation": "V1",
    "image_version": "0.0.1",
    "os_sku" : "Ubuntu",
    "os_version" : "18.04"
}

```

Comment (xuto2) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:46]: so os_sku and offer are duplicated and os_version and sku_name are duplicated.

**Review by xuto2 — APPROVED** [review]

File: vhdbuilder/packer/generate-vhd-publishing-info.sh (hunk 1 above)
Comment (hbeberman) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh]: Remove os_sku and os_version from the JSON and instead we now pass the value of os_sku through offer_name.

**Pull request merged**